crates/core/src/operations/writer.rs

```diff
@@ -427,10 +419,15 @@ impl PartitionWriter {
         Ok(())
     }
 
+    /// Retrieves the list of [Add] actions associated with the files written.
+    pub async fn drain_files_written(&mut self) -> Vec<Add> {
+        self.files_written.drain(..).collect()
+    }
+
     /// Close the writer and get the new [Add] actions.
     pub async fn close(mut self) -> DeltaResult<Vec<Add>> {
         self.flush_arrow_writer().await?;
-        Ok(self.files_written)
+        Ok(self.files_written.clone())
```

Comment:
self is moved here, so after calling close its not usable. This clone should be unnecessary.